ticketbuyer/purchase.go

```diff
@@ -65,4 +74,25 @@ out:
 			break out
 		}
 	}
+	p.wg.Done()
+}
+
+// Start starts the purchase manager goroutines.
+func (p *PurchaseManager) Start() {
+	p.wg.Add(1)
+	go p.NotificationHandler()
+}
+
+// WaitForShutdown blocks until all purchase manager goroutines have finished executing.
+func (p *PurchaseManager) WaitForShutdown() {
```

Comment:
This looks right but we need to actually use it.  Not called anywhere currently.